From aa53928ed7b763abd882f2a5efb8f98f53ccc6a9 Mon Sep 17 00:00:00 2001
From: Alex Vlasov <alex.m.vlasov@gmail.com>
Date: Sun, 20 Jun 2021 16:09:42 +0200
Subject: [PATCH] Squashed implementation of the pass

  Note: for Debian we only care about the failing armhf test which is fixed by
  a subset of the PR, namely what's below. For details see
  https://github.com/rust-lang/rust/pull/86383#issuecomment-864521922

--- a/library/std/src/thread/local/tests.rs
+++ b/library/std/src/thread/local/tests.rs
@@ -297,7 +297,7 @@
             .unwrap();
 
         loop {
-            match SYNC_STATE.compare_exchange_weak(
+            match SYNC_STATE.compare_exchange(
                 THREAD1_WAITING,
                 MAIN_THREAD_RENDEZVOUS,
                 Ordering::SeqCst,
